code/game/machinery/hivebot_swarm_core.dm

```diff
@@ -0,0 +1,111 @@
+/obj/machinery/hivebot_swarm_core //A hulking machine armed with several weapons. Creates hivebots and defends itself with a buzzsaw and mounted weapons.
+	name = "hivebot swarm core"
+	desc = "An incredibly dangerous machine with random weapons haphazardly attached with spot welds. Its main body seems impervious to damage, but the head is vulnerable."
+	icon = 'icons/effects/96x96.dmi'
+	icon_state = "hivebot_swarm_core"
+	pixel_x = -32
+	pixel_y = -16
+	use_power = FALSE
+	obj_integrity = 250
+	max_integrity = 250
+	anchored = TRUE
+	density = TRUE
+	var/recharge_period = 0 //The number of ticks the core is waiting for and cannot do anything
+	var/mob/living/threat_to_swarm //The core's "target" for its attacks
```

Comment:
The hivebots themselves are, but this big huge machine's sole purpose is to make more and to kill anything carbon-based that gets nearby. I like to headcanon that hivebots are 100% absorbed in efficiency and cut corners wherever possible, so it'd make sense that the hivebots themselves (who murder) can recognize threats, but the fabricator just recognizes the most obvious enemies.